squire_lib/src/accounts.rs

```diff
@@ -63,19 +76,33 @@ pub struct OrganizationAccount {
 }
 
 impl SquireAccount {
-    /// Creates a new account object
-    pub fn new(user_name: String, display_name: String) -> Self {
-        Self {
-            user_name,
+    pub fn create(user_name: String, display_name: String, permissions: SharingPermissions) -> Self {
+        SquireAccount {
             display_name,
-            user_id: Uuid::new_v4().into(),
-            arena_name: None,
-            mtgo_name: None,
-            trice_name: None,
-            do_share: SharingPermissions::Everything,
+            user_name,
+            gamer_tags: HashMap::new(),
+            user_id: UserAccountID::new(Uuid::new_v4()),
+            permissions
+        }
+    }
+
+    pub fn update_tags(&mut self, platfrom: Platforms, user_name: String) {
+        self.gamer_tags.insert(platfrom, user_name);
+    }
+
+    pub fn get_tags(&self, platforms: Vec<Platforms>) -> Vec<Option<String>> {
+        let tags = Vec::new()
+        for platform in platforms {
+            let gamer_tag = self.gamer_tags.get(platform);
+            tags.insert(Some(gamer_tag.clone()));
         }
+        tags
     }
 
+    pub fn delete_tags(&mut self, platform: Platforms) {
+        self.gamer_tags.insert(platform, None)
```

Comment:
We need to just remove this.
```suggestion
        self.gamer_tags.remove(platform);
```